vtex/faststore
# Feat: Adds jest-axe and Card component test

## What's the purpose of this pull request?

- Setup `@testing-library/jest-dom` to help to test our components when checking an element's attributes, its text content, its css classes...
- Adds `Card` component tests
- Introduces accessibility tests:  This won't guarantee that what we're building is accessible, but it's a first step towards it. 🥳 
- Updates `CheckboxField` component test 

TODO: Document 

## How it works?

Currently, our components library do not have tests, but we want to introduce them for upcoming components. For existing components, we will gradually add tests retroactively.

## How to test it?

1. navigate to `packages/components` -> run `yarn install`
2. run `yarn test`, you will be able to see the tests running
<img width="873" alt="image" src="https://github.com/user-attachments/assets/1e4c0fcc-4ded-45c5-a078-0fad99c52dba">

_____

3. removing `<Label>` component from `CheckboxField`, when running the step 2, you will get:
<img width="1005" alt="image" src="https://github.com/user-attachments/assets/898360ba-2a9a-40d5-afd4-a6c0f74d11a4">


## References

https://testing-library.com/docs/ecosystem-jest-dom
https://github.com/testing-library/jest-dom/issues/546
https://github.com/nickcolley/jest-axe/issues/79
